.github/workflows/python-test.yml

```diff
@@ -35,6 +35,7 @@ jobs:
     strategy:
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11"]
+        pydantic-version: ["pydantic_1", "pydantic_2"]
```

Comment:
It's great to see this coverage but I feel like this could significantly increase the build time since we have to test the combo of Python and Pydantic versions.

Reply:
Ah good point - especially since we don't need to test pydantic-v1 on all of the other python packages.

I can work on making another set of steps for testing the pydantic v1 functionality.

Reply:
I've added steps for testing this in the main build job and removed the extra matrix arguments